addons/mail/models/mail_link_preview.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from lxml import html
+import requests
+from odoo import api, models, fields
+
+class link_preview(models.Model):
+    _name = 'mail.link.preview'
+    _description = "Store link preview data's"
+
+    attachment_id = fields.Many2one('ir.attachment', 'Attachement previewed', index=True, ondelete='cascade')
+    type = fields.Char('type')
+    url = fields.Char('url')
+    title = fields.Char('title')
+    image = fields.Char('image')
+    description = fields.Text('description')
+
+    @api.model
+    def getOpenGraphData(self, url):
```

Comment:
we don't really use camelcase on python method, lower case with underscore to be consistent